tests/api_fastapi/execution_api/routes/test_xcoms.py

```diff
@@ -40,19 +41,21 @@ class TestXComsGetEndpoint:
     @pytest.mark.parametrize(
         ("value", "expected_value"),
         [
-            ("value1", "value1"),
-            ({"key2": "value2"}, {"key2": "value2"}),
-            ({"key2": "value2", "key3": ["value3"]}, {"key2": "value2", "key3": ["value3"]}),
-            (["value1"], ["value1"]),
+            ('"value1"', '"value1"'),
+            ('{"key2": "value2"}', '{"key2": "value2"}'),
+            ('{"key2": "value2", "key3": ["value3"]}', '{"key2": "value2", "key3": ["value3"]}'),
+            ('["value1"]', '["value1"]'),
```

Comment:
Now the contract between API Server and its client is simple: It expects JSON serialized string and will return string. 

It pushes down responsibility to serialize it into native (Python, Golang) format to the respective clients.

In one of the future PRs, I will move some of the serialization and deserialization logic in Python clients